pg_search/src/api/aggregate.rs

```diff
@@ -152,7 +152,7 @@ impl<'a> ParallelAggregationWorker<'a> {
         let nsegments = self.config.total_segments;
 
         let mut segment_ids = FxHashSet::default();
-        let (_, many_segments) = chunk_range(nsegments, nworkers, worker_number as usize);
+        let (_, many_segments) = chunk_range(nsegments, nworkers, worker_number.max(0) as usize);
```

Comment:
I'm not sure this is correct.

I think `worker_number` needs to instead become:

```rust
let worker_number = if pg_sys::parallel_leader_participation { worker_number + 1 } else { worker_number };
```

I think what you have written here would effectively have two "worker_number = 0"s checking out the same segments -- the leader and the real number 0.